public/population_submission/population_submission.rb

```diff
@@ -60,83 +68,85 @@ def create_record(class_name, data)
-    STDERR.puts "Encountered the following BIP request error."
-    STDERR.puts response['reason']
-    exit 1
-  end
-end
-
 
 puts " 1. Creating experimental plant_population "
 
+plant_population_id = create_record('plant_population',
+  name: 'RIPR_test_population',
+  description: '387 Brassica napus accessions used in the RIPR project for which mRNAseq data were submitted to SRA under PRJNA309367',
+  establishing_organisation: 'York University',
+  taxonomy_term_id: 27  #this is Brassica napus id in BIP
+)
 
-   create_record('plant_population',
-   name: 'RIPR_test_population',
-   description: '387 Brassica napus accessions used in the RIPR project for which mRNAseq data were submitted to SRA under PRJNA309367',
-   establishing_organisation: 'York University',
-   taxonomy_term_id: 27  #this is Brassica napus id in BIP
-   )
-
-
-puts ' 2. Submitting plant_varieties'
 
-       def record_plant_variety(plant_variety_name, crop_type)
-        request = Net::HTTP::Get.new("/api/v1/plant_varieties?plant_variety[query][plant_variety_name]=#{URI.escape plant_variety_name}", @headers)
-        response = call_bip request
-        if response['meta']['total_count'] == 0
-          create_record('plant_variety',
-          plant_variety_name: row[VARIETY],
-          crop_type: row[CROP_TYPE])
-        
-        else
-          response['plant_varieties'][0]['id']
-        end
-      end
-
-
-puts ' 3. Submitting plant_lines'
+# Function that finds or submits plant_varieties
+def record_plant_variety(plant_variety_name, crop_type)
+  request = Net::HTTP::Get.new("/api/v1/plant_varieties?plant_variety[query][plant_variety_name]=#{URI.escape plant_variety_name}", @headers)
+  response = call_bip request
+  if response['meta']['total_count'] == 0
+    create_record('plant_variety',
+      plant_variety_name: plant_variety_name,
+      crop_type: crop_type
+    )
+  else
+    response['plant_varieties'][0]['id']
+  end
+end
 
-      def record_plant_lines(plant_line_name, plant_variety_name)
-        request = Net::HTTP::Get.new("/api/v1/plant_linesplant_line[query][plant_line_name]=#{URI.escape plant_line_name}", @headers)
-        response = call_bip request
-        if response['meta']['total_count'] == 0
-          create_record('plant_line',
-          Plant_line_name: row[LINE_NAME],
-          plant_variety_name: row[VARIETY])
-        else
-          response['plant_lines'][0]['id']
-        end
-      end
 
-puts '4. Submitting plant_accessions'
+# Function that finds or submits plant_lines
+def record_plant_line(plant_line_name, plant_variety_id, comments)
+  request = Net::HTTP::Get.new("/api/v1/plant_lines?plant_line[query][plant_line_name]=#{URI.escape plant_line_name}", @headers)
+  response = call_bip request
+  if response['meta']['total_count'] == 0
+    create_record('plant_line',
+      plant_line_name: plant_line_name,
+      plant_variety_id: plant_variety_id,
+      comments: comments #SRA identifier
```

Comment:
yes, sequence_identifier is the appropriate location. Sorry even though you talked about it here before, I wasn't using it in the submission. please migrate the identifiers over from PL.comments to sequence_identifiers. I will alter the script accordingly.